File: stack/1544_make_the_string_great/1544_make_the_string_great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        for i in s:
            if stack and ((ord(stack[-1])-32) == ord(i) or (ord(stack[-1])+32) == ord(i)):
                stack.pop()
            else:
                stack.append(i)
        return ''.join(stack)


s = "leEeetcode"
res = Solution().makeGood(s)
print(res)

File: stack/1544_make_the_string_great/1544_make_the_string_great.go
package main

import "fmt"

func makeGood(s string) string {
	ans := []byte{}
	for i := 0; i < len(s); i++ {
		ans = append(ans, s[i])
		size := len(ans)
		if size >= 2 {
			if mequal(ans[size-1], ans[size-2]) {
				ans = ans[:size-2]
			}
		}
	}
	return string(ans)
}

func mequal(a, b byte) bool {
	return (a ^ ' ') == b
}

func main() {
	s := "leEeetcode"
	res := makeGood(s)
	fmt.Println(res)
}
